include/uapi/sound/sof-ipc.h

```diff
@@ -476,6 +476,7 @@ struct sof_ipc_stream_params {
 	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
 	enum sof_ipc_chmap chmap[SOF_IPC_MAX_CHANNELS];	/* channel map */
+	uint32_t private_data[64];
```

Comment:
Did you mean 'reserved'. Private data means you are using it in a opaque way, reserved means don't mess with this structure, it's for future usages.

Reply:
it is private for HDA.  Please see comment below

Reply:
it's ugly, there is no reason to make things opaque. better use a union as done for the SSP/dmic.
Also you are not getting the point. 'reserved' means not currently used but may be reclaimed in future versions. It's completely different from private or HDAudio specific where you do make use of this. You do not need 64 bytes to handle HDaudio, but it would make sense to add 64 bytes of reserved data to be future-proof.